pkg/resources/scim_integration.go

```diff
@@ -17,40 +20,49 @@ var scimIntegrationSchema = map[string]*schema.Schema{
 		Type:        schema.TypeString,
 		Required:    true,
 		ForceNew:    true,
-		Description: "Specifies the name of the SCIM integration. This name follows the rules for Object Identifiers. The name should be unique among security integrations in your account.",
+		Description: "String that specifies the identifier (i.e. name) for the integration; must be unique in your account.",
+	},
+	"enabled": {
+		Type:        schema.TypeBool,
+		Required:    true,
+		Description: "Specify whether the security integration is enabled.",
 	},
 	"scim_client": {
 		Type:        schema.TypeString,
 		Required:    true,
+		ForceNew:    true,
 		Description: "Specifies the client type for the scim integration",
-		ValidateFunc: validation.StringInSlice([]string{
-			"OKTA", "AZURE", "GENERIC",
-		}, true),
-		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-			normalize := func(s string) string {
-				return strings.ToUpper(strings.ReplaceAll(s, "-", ""))
-			}
-			return normalize(old) == normalize(new)
-		},
+		ValidateFunc: validation.StringInSlice(sdk.AsStringList([]sdk.ScimSecurityIntegrationScimClientOption{
+			sdk.ScimSecurityIntegrationScimClientOkta, sdk.ScimSecurityIntegrationScimClientAzure, sdk.ScimSecurityIntegrationScimClientGeneric,
+		}), true),
+		DiffSuppressFunc: ignoreTrimSpaceSuppressFunc,
 	},
-	"provisioner_role": {
+	"run_as_role": {
 		Type:        schema.TypeString,
 		Required:    true,
+		ForceNew:    true,
 		Description: "Specify the SCIM role in Snowflake that owns any users and roles that are imported from the identity provider into Snowflake using SCIM.",
-		ValidateFunc: validation.StringInSlice([]string{
-			"OKTA_PROVISIONER", "AAD_PROVISIONER", "GENERIC_SCIM_PROVISIONER",
-		}, true),
-		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-			normalize := func(s string) string {
-				return strings.ToUpper(strings.ReplaceAll(s, "-", ""))
-			}
-			return normalize(old) == normalize(new)
-		},
+		ValidateFunc: validation.StringInSlice(sdk.AsStringList([]sdk.ScimSecurityIntegrationRunAsRoleOption{
+			sdk.ScimSecurityIntegrationRunAsRoleOktaProvisioner, sdk.ScimSecurityIntegrationRunAsRoleAadProvisioner, sdk.ScimSecurityIntegrationRunAsRoleGenericScimProvisioner,
+		}), true),
+		DiffSuppressFunc: ignoreTrimSpaceSuppressFunc,
```

Comment:
casing in the previous suppression mattered

Reply:
Fixed. According to docs: 
> The values OKTA_PROVISIONER, AAD_PROVISIONER, and GENERIC_SCIM_PROVISIONER are case-sensitive and must always be capitalized.

Reply:
I was not clear with the previous comment:
1. Yes, they must be capitalized and we handle it on the SDK level.
2. We agreed that we allow users to put e.g. lower cased values (this was the previous behavior).
3. With the old implementation (before your changes) it was allowed, and diff suppression handled also lowercase correctly (and my comment was only about that using `ignoreTrimSpaceSuppressFunc` breaks it)

Reply:
Reverted to old implementation.

Reply:
Hmm. Okay, but the old implementation (with `return strings.ToUpper(strings.ReplaceAll(s, "-", ""))` is kinda fishy, is it necessary?

Let's talk about this and solve this in the next PR.